### ISSUE TITLE
Positioning of injected code

### ISSUE THREAD

**Issue by iamtompickering** [nystudio107/craft-seomatic#1286]
### Question
My client has requested that we move the meta and title tags so that they appear at the very beginning of the head. 

Alongside this, they would also like to move the JSON schema to sit right after these and also move the JS e.g. GA to the bottom of the page. 

Is any of this possible and if so please can you let me know how to implement this?

**Comment by khalwat** [comment]
You don't want the JSON schema to appear in the `<head>` -- it should be where it is now, before the `</body>` tag.

The JS for GA already appears where Google says it should go. Depending on the GA script, a small stub will appear in the `<head>` which loads the GA script asynchronously (in a non-blocking way), and some of them may also have a `<script>` tag in the `<body>` again, both at Google's recommendations.

TL;DR It's already doing what Google suggests, and what are best practices from a performance point of view.

Can you elaborate on the reasons for their proposed changes?

**Comment by iamtompickering** [comment]
Please could you just explain why the JSON should be before the `</body>` tag for me? Client is just questionning this and have said the following

```
The reason why I want the JSON Schema in the head section is due to HTML5 specification. 

As the head element represents a collection of metadata for the Document.
Source: https://www.w3.org/TR/2014/REC-html5-20141028/document-metadata.html#the-head-element 

It makes more sense to have it place in the head tag.
```

**Comment by khalwat** [comment]
Because you generally want the `<head>` to be as lightweight as possible for performance reasons. Remember, the only things that care about JSON-LD are bots, not users:

https://youtu.be/MHyAOZ45vnU

&&

https://www.smashingmagazine.com/2021/09/css-head-tag/

Browsers can load the page more efficiently if the head is not monstrously large, and JSON-LD data can tend to be rather large.

Google says JSON-LD can go in the `<head>` or in the `<body>` tag:

https://developers.google.com/search/docs/appearance/structured-data/intro-structured-data#supported-formats

> **JSON-LD** (Recommended) | A JavaScript notation embedded in a <script> tag in the <head> and <body> elements of an HTML page. The markup is not interleaved with the user-visible text, which makes nested data items easier to express, such as the Country of a PostalAddress of a MusicVenue of an Event. Also, Google can read JSON-LD data when it is dynamically injected into the page's contents, such as by JavaScript code or embedded widgets in your content management system.

FWIW, Yoast also puts the JSON-LD in the `<body>`

**Comment by khalwat** [comment]
I'm gonna guess these instructions are coming from an SEO agency or "SEO person"?

**Comment by iamtompickering** [comment]
@khalwat brilliant thanks for all of this, I will relay it back to them. 

And yes these instructions are indeed coming from an SEO person. Not my expertise, so I'm very thankful for the  explanations here!

**Comment by khalwat** [comment]
Condolences! :)
